apps/studio/src/features/editing-experience/components/FormBuilder.tsx

```diff
@@ -0,0 +1,166 @@
+import {
+  Box,
+  FormControl,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInputField,
+  NumberInputStepper,
+  RadioGroup,
+  Select,
+} from '@chakra-ui/react'
+import {
+  Button,
+  FormLabel,
+  Input,
+  NumberInput,
+  Radio,
+  Switch,
+} from '@opengovsg/design-system-react'
+import { IsomerComponent } from '@opengovsg/isomer-components'
+import { useForm } from 'react-hook-form'
+import IsomerSchema from './0.1.0.json'
+import { getSchemaFieldType } from '~/utils/schema'
+
+// TODO: Props here might not make sense, need to check again
+export interface FormBuilderProps {
+  component: keyof typeof IsomerSchema.components.complex
+  data?: IsomerComponent
+}
+
+// Strategy:
+// - Read the sub-schema for the given component
+// - For each prop inside the sub-schema, render the appropriate input field
+//
+// TODO:
+// - Support type = 'array', format = 'prose' to be Tiptap editor
+// - Support type = 'array' with items of type = 'object' to be sub-form (no drag and drop yet)
+// - Support $ref to other parts of the schema
+// - Handle grouping of fields
+export const FormBuilder = ({
+  component,
+  data,
+}: FormBuilderProps): JSX.Element => {
+  const schema = IsomerSchema.components.complex[component]
+  const { required, properties } = schema
+  const { register, handleSubmit } = useForm<typeof properties>()
+
+  // Utility type to get all keys from a union of objects
+  type UnionKeys<T> = T extends T ? keyof T : never
+  type ComplexComponentProps = UnionKeys<typeof properties>
```

Comment:
nit: shift types outside

Reply:
Updated in 5374fc108bd5295f6aa946227be03ec544c95ac9.